cypress/e2e/environment.cy.js

```diff
@@ -0,0 +1,389 @@
+          cy.get(
+            parameter.name === "Branch"
+              ? '[aria-label="repo_branch-input"]'
+              : '[aria-label="repo_url-input"]'
+          )
+            .clear()
+            .type("New Value " + parameter.name);
+          break;
+      }
+      if (parameter.name !== "Repository" && parameter.name !== "Branch") {
+        cy.get('[aria-label="' + parameter.name + '-submit-edit"]').click();
+        cy.get('[aria-label="' + parameter.name + '-value"]').should(
+          "contain",
+          "New Value " + parameter.name
+        );
+      } else {
+        cy.get('[aria-label="Repository Settings-submit-edit"]').click();
+        if (parameter.name === "Repository") {
+          cy.get('[aria-label="repo_url-value"]').should(
+            "contain",
+            "New Value " + parameter.name
+          );
+        } else {
+          cy.get('[aria-label="repo_branch-value"]').should(
+            "contain",
+            "New Value " + parameter.name
+          );
+        }
+      }
+    });
+    deleteEnv(4);
+  });
+
+  it("1.5 Clear environment", function () {
+    //Fill The form and submit
+    cy.visit("/console/");
+    cy.get('[aria-label="Environment card"]').contains("test").click();
+    cy.get('[aria-label="ServiceCatalog-Success"]').should("to.be.visible");
+
+    //Go to settings
+    cy.get('[aria-label="Settings actions"]').click();
+    cy.url().should("contain", "/console/settings?env=");
+    cy.get('[aria-label="Name-value"]').should("contain", "test");
+
+    //Cancel Clear Env and expect nothing to change
+    cy.get("button").contains("Clear environment").click();
+    cy.get("button").contains("Cancel").click();
+    cy.visit("/console/");
+    cy.get('[aria-label="Environment card"]').contains("test").click();
+    cy.get('[aria-label="ServiceCatalog-Success"]').should("to.be.visible");
+
+    //Go to settings
+    cy.get('[aria-label="Settings actions"]').click();
+    cy.url().should("contain", "/console/settings?env=");
+    cy.get('[aria-label="Name-value"]').should("contain", "test");
+
+    //Clear Env
+    cy.get("button").contains("Clear environment").click();
+    cy.get('[aria-label="clear environment check"]').type("test");
+    cy.get("button")
+      .contains("I understand the consequences, clear this environment")
+      .click();
+    cy.visit("/console/");
+    cy.get('[aria-label="Environment card"]').contains("test").click();
+    cy.get('[aria-label="ServiceCatalog-Empty"]').should("to.be.visible");
+    //Update service catalog to restore instances
+    cy.get("button").contains("Update Service Catalog").click();
+    cy.get("button").contains("Yes").click();
+    cy.intercept(
+      "/api/v2/environment/a5f9f7f2-5f43-4fad-a16d-f71f127f0e91?details=fals"
+    ).as("catalogUpdate");
+    cy.wait("@catalogUpdate")
+      .its("response.body")
+      .then(() => {
+        cy.get('[aria-label="ServiceCatalog-Success"]').should("to.be.visible");
```

Comment:
that's the toaster, is I'm not mistaken, but we also need to assert that the services are back in the list :)

Reply:
that's label corresponds to the component that holds these services, if you think that I should also assert toast, I'll add it

Reply:
Maybe it would be good to check if the length of the list of services is the same as before clearing the environment 🤔

Reply:
As I mentioned through the call(but don't want to lost that knowledge so I'll leave it here :) ), that would cause issues, as sometimes we got only part of the services when the status is changed, the rest is added on second call, which is problematic as there is no simple way to assert that in some stable enough way. At least not to my knowledge at the moment. We can come back to this test case in the future.